src/dlfcn.c

```diff
@@ -467,6 +469,169 @@ char *dlerror( void )
     return error_buffer;
 }
 
+/* taken from http://bandido.ch/programming/Import_Address_Table_Hooking.pdf */
+IMAGE_IMPORT_DESCRIPTOR* getImportTable( HMODULE module, DWORD *size )
+{
+    IMAGE_DOS_HEADER* dosHeader = (IMAGE_DOS_HEADER*)module;
+    if ( dosHeader->e_magic != 0x5A4D )
+        return NULL;
+    IMAGE_OPTIONAL_HEADER* optionalHeader = (IMAGE_OPTIONAL_HEADER*)
+        ((BYTE*)module + dosHeader->e_lfanew + 24);
+    if ( optionalHeader->Magic != IMAGE_NT_OPTIONAL_HDR_MAGIC )
+        return NULL;
+    if ( optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].Size == 0 ||
+        optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress == 0 )
+        return NULL;
+    *size = optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].Size;
+    return (IMAGE_IMPORT_DESCRIPTOR*)((BYTE*)module +
+        optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress);
+}
+
+#ifdef _WIN64
+typedef ULONGLONG PointerType;
+#else
+typedef ULONG PointerType;
+#endif
+
+/*
+ * return symbol name for a given address
+ */
+char *getSymbolName( HMODULE baseAddress, IMAGE_IMPORT_DESCRIPTOR *iid, const void *addr )
+{
+    PointerType base = (PointerType)baseAddress;

```

Comment:
Just use `void *` and then there is no need for ifdef-ed `PointerType` type.

Reply:
With the requested change I get with gcc cross compiler 8.2.0 the following warning 
```
/home/ralf/src/dlfcn-win32/src/dlfcn.c: In function 'getSymbolName':
/home/ralf/src/dlfcn-win32/src/dlfcn.c:497:24: warning: initialization of 'PointerType' {aka 'long unsigned int'} from 'HMODULE' {aka 'struct HINSTANCE__ *'} makes integer from pointer without a cast [-Wint-conversion]
  497 |     PointerType base = baseAddress;
      |                        ^~~~~~~~~~~
```

Reply:
With the requestd change on 32 bit I get:
```
/home/ralf/src/dlfcn-win32/src/dlfcn.c: In function 'getSymbolName':
/home/ralf/src/dlfcn-win32/src/dlfcn.c:506:39: warning: comparison between pointer and integer
  506 |             if (thunkIAT->u1.Function != (PointerType)addr)
      |                                       ^~
```

Reply:
If you use `void*` type instead of `PointerType` type then you should not get anything related to keyword `PointerType` in warning message.

Reply:
Why you cannot use directly `if (thunkIAT->u1.Function != baseAddress)`?

Reply:
> With the requestd change on 32 bit I get:
> 
> ```
> /home/ralf/src/dlfcn-win32/src/dlfcn.c: In function 'getSymbolName':
> /home/ralf/src/dlfcn-win32/src/dlfcn.c:506:39: warning: comparison between pointer and integer
>   506 |             if (thunkIAT->u1.Function != (PointerType)addr)
>       |                                       ^~
> ```

Cannot this `if ((void *)thunkIAT->u1.Function != addr)` help?

Reply:
Yes, that helps, thanks.

Reply:
Looks like there is missing fixup change for this part...